doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,67 @@
+# Troubleshooting Production Issues related to Unsupported Operations in Mountpoint
+
+Mountpoint enables users to access S3 buckets through local filesystem. While this allows for existing workflows and applications to work with S3 unchanged, it also limits the effective API for Mountpoint to be a subset of the filesystem API.
+So, there are a few operations in filesystem API which are not supported using Mountpoint on a mounted bucket. Here, we are enumerating those common cases with their errors reported in the logs which customers face. Please look at [SEMANTICS document](doc/SEMANTICS.md) for more information on what operations are allowed in Mountpoint.
+
+## Non-sequential Write
+Mountpoint only supports sequential writes. On random out of order writes users should expect `Invalid Argument`. For example - 
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+In the logs, we get the following WARN message - 
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed:\
+upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
```

Comment:
Let's format the markdown with more spacing around headers and code blocks.

Also, we should prefer passive voice in the documentation where possible.

````suggestion
## Random writes

Mountpoint supports writing to a file sequentially. Random writes, or 'out-of-order' writes, will return an error to FUSE, which may appear in applications with the error message "Invalid argument".

For example, the following code seeks one byte into the file and writes a single byte using `dd`.

```
$ dd if=/dev/random of=out seek=1 count=1
dd: writing to 'out': Invalid argument
```

In Mountpoint's logs, a warning message will be emitted similar to below:

```
WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed: upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
```
````